dotnet/src/Agents/Core/AgentGroupChat.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Agents.Chat;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace Microsoft.SemanticKernel.Agents;
+
+/// <summary>
+/// A an <see cref="AgentChat"/> that supports multi-turn interactions.
+/// </summary>
+public sealed class AgentGroupChat : AgentChat
+{
+    private readonly HashSet<string> _agentIds; // Efficient existence test
+    private readonly List<Agent> _agents; // Maintain order (fwiw)
+
+    /// <summary>
+    /// Indicates if completion criteria has been met.  If set, no further
+    /// agent interactions will occur.  Clear to enable more agent interactions.
+    /// </summary>
+    public bool IsComplete { get; set; }
```

Comment:
You mean automatically clear the flag on re-invocation?  That would be effecient although it does enable a blind continuation of the chat (say if the application code was looping and not regarding the completion flag...I suppose that would be a coding bug.)

I'm happy to go in this direction, if we're all fine w/ a more agressive execution assumptions.

@markwallace-microsoft / @matthewbolanos ???

Reply:
I went ahead and introduced an option to manually reset completion, and the default is `false` (don't automatically reset) and modified behavior to throw instead of silently yield if already complete.